packages/expo-location/src/Location.ts

```diff
@@ -159,8 +171,39 @@ export async function enableNetworkProviderAsync(): Promise<void> {
 }
 
 export async function getCurrentPositionAsync(
-  options: LocationOptions = {}
+  options: LocationOptions = {},
+  methodOptions: LocationMethod = { type: LocationMethodType.Fast }
 ): Promise<LocationData> {
+  console.log(methodOptions);
+  if (methodOptions.type === LocationMethodType.Fast) {
+    console.log("fast");
+    return new Promise<LocationData>(async (resolve, reject) => {
+      try {
+        let bestPosition: LocationData | null = null;
+        let current = 0;
+        let { tries, minAccuracyRadius } = {
+          tries: 1,
+          minAccuracyRadius: 20000,
+          ...methodOptions,
+        };
+        let { remove } = await watchPositionAsync(options, position => {
+          current++;
+          if (bestPosition === null) {
+            bestPosition = position;
+          } else if (bestPosition.coords.accuracy > position.coords.accuracy) {
+            bestPosition = position;
+          }
+          if (current >= tries || bestPosition.coords.accuracy <= minAccuracyRadius) {
+            remove();
+            resolve(bestPosition);
+          }
+        });
+      } catch (e) {
+        reject(e);
+      }
+    });
+  }
+  console.log("slow");
   return ExpoLocation.getCurrentPositionAsync(options);
```

Comment:
I believe it would be nice to have your changes extracted to separate function with some TSDoc saying that:
```ts
/**
 * hey, I'm going to find out local position by subscribing to position events and when 
 * conditions are satisfied I would return the results
 */
```